newIDE/app/src/ResourcesList/ResourceUtils.js

```diff
@@ -80,8 +80,24 @@ export const copyAllToProjectFolder = (
         return resourcePath;
       }
 
-      const resourceBasename = path.basename(resourcePath);
-      const resourceNewPath = path.join(projectPath, resourceBasename);
+      const resourceBasename = path.basename(resourcePath),
+        splitAt = resourceBasename.lastIndexOf('.'),
+        originalFileName = resourceBasename.substring(0, splitAt),
+        fileExtension = resourceBasename.substring(splitAt),
+        allResourcesNames = project.getResourcesManager().getAllResourceNames();
+
+      let fileName = originalFileName,
+        fileNumber = 1;
+
+      for (let itr = 0; itr < allResourcesNames.size(); itr++) {
+        if (fileName + fileExtension === allResourcesNames.at(itr)) {
+          fileName = originalFileName + fileNumber;
+          fileNumber++;
+          itr = 0;
+        }
+      }
```

Comment:
Can you investigate using `newNameGenerator`? Search for it in the rest of the codebase.